src/decrypt.c

```diff
@@ -91,19 +97,27 @@ ece_decrypt_records(const uint8_t* key, const uint8_t* nonce, uint32_t rs,
   size_t recordStart = 0;
   size_t blockStart = 0;
   for (size_t counter = 0; recordStart < ciphertextLen; counter++) {
-    size_t recordEnd = recordStart + rs;
-    if (recordEnd > ciphertextLen) {
+    size_t recordEnd;
+    if (rs > ciphertextLen - recordStart) {
+      // This check is equivalent to `recordStart + rs > ciphertextLen`; it's
```

Comment:
Hm, this might not entirely the same. I'm not entirely sure how this is handled, but ciphertextLen - recordStart could be negative, which would overflow if it's handled as an unsigned. So maybe checking that ciphertextLen > recordStart in addition would make sense.